snooty/n.py

```diff
@@ -338,7 +339,7 @@ class Role(InlineParent):
 class RefRole(Role):
     __slots__ = ("fileid", "url")
     type = "ref_role"
-    fileid: Optional[str]
+    fileid: Optional[Tuple[str, Optional[str]]]
```

Comment:
Just curious about why it's preferable to use a tuple here rather than add an `html_id` field!

Reply:
Yes! So let me know if you think this is silly and unnecessary, but this lets us encode in the type system that the `html_id` can be set if and only if the `fileid` is set! Except, this type signature can actually be changed to `Optional[Tuple[str, str]]`, which would say "fileid and html_id must both be set, or neither are set".

If they were separate fields, we couldn't have that kind of dependent typing.

Reply:
No, that's a great point! At first I thought this would add unnecessary complexity to the front end, but that all-or-nothing approach makes things pretty clear. And changing the type signature to `Optional[Tuple[str, str]]` seems even better!